File: php/grade-school/grade-school.php
<?php

class School {
  private $studentsByGrade = [];

  public function add($name, $grade) {
    $this->studentsByGrade[$grade][] = $name;
    sort($this->studentsByGrade[$grade]);
  }

  public function grade($grade) {
    if (empty($this->studentsByGrade[$grade])) {
      return [];
    }
    return $this->studentsByGrade[$grade];
  }

  public function studentsByGradeAlphabetical() {
    ksort($this->studentsByGrade);
    return $this->studentsByGrade;
  }
}

File: clojure/grade-school/src/grade_school.clj
(ns grade-school)

(defn grade [school grade]
  (get school grade []))

(defn add [school name grade]
    (merge school {grade (vec (conj (school grade) name))}))

(defn sorted [school]
  (->> school
       (seq)
       (map (fn [[grade students]] [grade (sort students)]))
       (sort)
       (into {})))